tests/integration/security/optional_jwt_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security
+
+import (
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/tmpl"
+	"istio.io/istio/tests/common/jwt"
+	"istio.io/istio/tests/integration/security/util"
+	"istio.io/istio/tests/integration/security/util/authn"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+const (
+	authHeaderKey = "Authorization"
+)
+
+// Test authentication policy with optional JWT, and enforce requirement per host, path etc via authZ.
+func TestOptionalAuthnJwt(t *testing.T) {
```

Comment:
There is already a TestV1beta1_JWT in `tests/integration/security/rbac_v1beta1_test.go` that covers the `originIsOptional: true`, could you merge this to the other one?

Reply:
Didn't realize we already have it. I've changed that test to use namespace-level policy and add an instance (`c`) that doesn't have authz to verify the JWT requirements won't be enforced on `c`.